sklearn/utils/extmath.py

```diff
@@ -692,3 +692,25 @@ def _batch_mean_variance_update(X, old_mean, old_variance, old_sample_count):
     return ((old_sum + new_sum) / updated_sample_count,
             unnormalized_variance / updated_sample_count,
             updated_sample_count)
+
+
+def eigen_sign_flip(u):
```

Comment:
I would rename this to `_deterministic_vector_sign_flip` (the `_` to keep it a private utility function). This function does not need to work only with eigen vectors of a matrix but any vectors so there is no need to put 'eigen' in the name or docstring.